pkg/datapath/linux/node.go

```diff
@@ -910,9 +913,28 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *nodeTypes.Node, firstAdd
 		if n.subnetEncryption() {
 			n.enableSubnetIPsec(n.nodeConfig.IPv4PodSubnets, n.nodeConfig.IPv6PodSubnets)
 		}
+
 		return nil
 	}
 
+	if option.Config.EnableWireguard {
+		var podCIDRv4, podCIDRv6 *net.IPNet
+		wgIPv4 := newNode.GetIPByType(addressing.NodeWireguardIP, false)
+		wgIPv6 := newNode.GetIPByType(addressing.NodeWireguardIP, true)
+		if newNode.IPv4AllocCIDR != nil {
+			podCIDRv4 = newNode.IPv4AllocCIDR.IPNet
+		}
+		if newNode.IPv6AllocCIDR != nil {
+			podCIDRv6 = newNode.IPv6AllocCIDR.IPNet
+		}
+		if err := n.wgAgent.UpdatePeer(newNode.Name, newNode.WireguardPubKey,
```

Comment:
Could it be the case that `Config.EnableWireguard` is true, but the agent failed to initialize and this will cause the agent to crash? Should we check for `n.Agent == nil` here?

Reply:
We will fatal if the WG agent failed to initialize:
```
wgAgent, err = wireguard.NewAgent(privateKeyPath,
			option.Config.WireguardSubnetV4, option.Config.WireguardSubnetV6)
		if err != nil {
			log.WithError(err).Fatal("Failed to initialize wireguard")
		}
```